src/components/PostDetails.tsx

```diff
@@ -5,31 +5,39 @@ import { NewCommentForm } from './NewCommentForm';
 import * as commentsApi from '../api/comments';
 
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { useAppSelector } from '../app/hooks';
+import { useDispatch } from 'react-redux';
+import {
+  addCurrentComment,
+  deleteCurrentComment,
+  setComments,
+  setLoaded,
+} from '../features/comments';
+import { setError } from '../features/comments';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useDispatch();
+  const { items, loaded, hasError } = useAppSelector(state => state.comments);
   const [visible, setVisible] = useState(false);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    dispatch(setLoaded(false));
+    dispatch(setError(false));
     setVisible(false);
 
     commentsApi
       .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+      .then(commentsFromServer => dispatch(setComments(commentsFromServer))) // save the loaded comments
+      .catch(() => dispatch(setError(true))) // show an error when something went wrong
+      .finally(() => dispatch(setLoaded(true))); // hide the spinner
```

Comment:
It seems like you're not handling the error case properly. If the API call fails, you're just setting the error state, but you're not giving any information about what went wrong. It would be helpful to also log the error or show an error message to the user.